generator/json-generator.lua

```diff
@@ -1,135 +1,153 @@
+local json = require 'lunajson'
+
 -- ------------------------------------------------
 -- Constants
 -- ------------------------------------------------
 
-local APOSTROPHE =  '"'
-local COMMA      =  ','
-local LINE_BREAK = '\n'
-local TAB        = '  '
-
-local PARAM  = '${%d:%s (%s)}'
-
 local OUTPUT_FILE = 'love-completions.json'
-
 local WIKI_URL = 'https://love2d.org/wiki/'
 
+local LOVE_MODULE_STRING = 'love.'
+
 -- ------------------------------------------------
 -- Local Functions
 -- ------------------------------------------------
 
-local function generateArguments( arguments, start )
-    local params = ''
-    start = start or 0
-
-    if arguments then
-        for i, args in ipairs( arguments ) do
-            params = params .. string.format( PARAM, start + i, args.name, args.type )
-
-            -- Add a separator unless we are dealing with the last argument of the function.
-            if i ~= #arguments then
-                params = params .. ', '
-            end
+---
+-- Generates a sequence containing subtables of all the arguments
+--
+local function createArguments( args )
+    local arguments = {}
+    for i, v in ipairs( args ) do
+        arguments[i] = {}
+        arguments[i].name = v.name
+
+        -- Use [foo] notation as display name for optional arguments.
+        if v.default then
+            arguments[i].displayName = string.format( '[%s]', v.name )
         end
     end
+    return arguments
+end
 
-    return params
+local function determineType( type )
+    if type == 'string'
+    or type == 'number'
+    or type == 'table'
+    or type == 'boolean'
+    or type == 'function' then
+        return type
+    end
+    return 'ref'
 end
 
-local function cleanUpString( str )
-    return str:gsub( '\n\n', ' ' ):gsub( '\"', '\\"' ):gsub( '\0', '' )
+local function createReturnTypes( returns )
+    local returnTypes = {}
+    for i, v in ipairs( returns ) do
+        returnTypes[i] = {}
```

Comment:
And delete the two lines that follow this line.

Also I would consider renaming this function to `getType` or something like that